### PR TITLE
Navigation: Integrate Explore actions into topnav

### LINKED ISSUE #56269: Explore: Integrate actions into topnav toolbar

### PR DESCRIPTION
**Pull request by ashharrison90**
**What this PR does / why we need it**:

- moves Explore actions into topnav when not split

# topnav on

https://user-images.githubusercontent.com/20999846/194050605-3b872d91-0680-46c0-9b13-355c9a162a83.mp4

# topnav off

https://user-images.githubusercontent.com/20999846/194050617-fa6e1dea-b49a-4f47-90fa-b328e01df756.mp4

**Which issue(s) this PR fixes**:

Fixes #56269 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by torkelo** [comment]
Noticed a bug sometimes the ToolbarButton menu is shown but without anything in it 
Screenshot from 2022-10-05 15-49-53

The close split feels like something that maybe should not be collapsed into the menu first. I think its position is a bit strange in the current design. Normally a close pane action is on the upper right corner of that "pane/window". But maybe tweaking button order should be done in a separate step. 

Screenshot from 2022-10-05 15-52-27

**Comment by ashharrison90** [comment]
have fixed the bug (i think!), but haven't moved any buttons.

lemme know if that's something we think we should do in this PR (or later, or not at all) 👍

**Comment by gelicia** [comment]
I agree that the close split button shouldn't be collapsed with other options right away. 

Another thing though, we will be adding split resizing soon that will add a minimize/maximize button in the split view  so people can expand/(partially) collapse their panes. I can be sure this works with top nav as well. I will plan on collapsing it with the same logic as the "Add To Dashboard" button. If you minimize a pane, it will be 200px wide, so it will likely be collapsed all the way anyway. We also had made the panes go horizontally scrollable at 600px to try to keep the query editors from looking squished, and alongside that, made the collapsed kebab menu left justified. This was because there was no easy way to keep the toolbar outside of the horizontal scroll. You can see what I'm talking about on this branch https://github.com/grafana/grafana/tree/kristina/split-resize-3 

We're going to be reworking this so any feedback would be helpful! Feel free to add it here https://github.com/grafana/grafana/issues/56340

**Comment by ashharrison90** [comment]
@gelicia pretty sure left-aligning that menu will mean the overflow behaviour won't work correctly. thoughts on just disabling split view under 600px (or w/e the actual breakpoint is)? i can't see how it will ever really be usable at that size tbh 🤔

File: public/app/features/explore/ExploreToolbar.tsx [hunk 1]
```diff
@@ -107,51 +116,123 @@ class UnConnectedExploreToolbar extends PureComponent<Props> {
...
-          pageIcon={exploreId === ExploreId.left ? 'compass' : undefined}
-          leftItems={[
-            exploreId === ExploreId.left && (
-              <DashNavButton
-                key="share"
-                tooltip="Copy shortened link"
-                icon="share-alt"
-                onClick={this.onCopyShortLink}
-                aria-label="Copy shortened link"
+    return [
```
[the remark this answers is not in the document]
Comment (ashharrison90) [reply public/app/features/explore/ExploreToolbar.tsx:143]: thanks @ifrost that's much nicer 👍 

agreed it's a little unwieldy at the moment 😬 will be a little cleaner again once the `topnav` toggle is turned on by default 🙂

**Review by ifrost — CHANGES_REQUESTED** [review]
Played around with split views and live mode with/without topnav and noticed a few things:

<details><summary>(1) When the user switches to a data source supporting live streaming the "live" button does not show up (though there's space for it and it appears upon page refresh). This happens with topnav=true and topnav=false (video in details section ⬇️ )</summary>

*main*

https://user-images.githubusercontent.com/745532/195034717-3dac17ff-b608-452e-8b72-63dd1d820b6b.mov

*branch*

https://user-images.githubusercontent.com/745532/195034146-01e68696-c9c3-43a1-a65c-628b6c5562e5.mov
</details>

(2) It seems we don't have borders around buttons in split view any more. Is that intended?

| main | branch |
|------|--------|
|  Screen Shot 2022-10-11 at 10 03 18 |  Screen Shot 2022-10-11 at 10 03 25   |

Other than that seems to work fine! 👍

**Comment by ashharrison90** [comment]
@ifrost thanks 🙌 i've fixed the bug with `Live` not appearing

`ToolbarButton`s not having borders around them when `topnav` is enabled [was a conscious choice](https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/ToolbarButton/ToolbarButton.tsx#L193) and isn't a result of this PR (turning on `topnav` in main should show the same behaviour).

**but** pretty sure buttons in the top bar were the only consideration, not sure how we can handle `ToolbarButton`s being used in other places gracefully 🤔 

wdyt @torkelo?

**Review by Elfo404 — APPROVED** [review]
LGTM.

a side note regarding the toolbar buttons: i think it looks ok also when in split mode, but i'm wondering if its intended to be used outside of the navbar at this point, it may even be considered a breaking change, afaik bot the the trace panel and the logs panel use that component in explore (@grafana/observability-logs  @grafana/observability-traces-and-profiling  you may want to double check as well).

**Comment by ashharrison90** [comment]
@Elfo404 yep agreed, think we need to have a conversation about our expectations for where `ToolbarButton` is used / how it should look consistently. might be that the easiest thing to do is to put the `ToolbarButton` styling back to what it was and override it specifically in the top navbar components 🤔

**Comment by ashharrison90** [comment]
@ifrost @Elfo404 not sure what's going on with the `Observability Experience test code coverage` step. i don't think it's related to the changes, but you might know better than me? 🤔

**Comment by Elfo404** [comment]
@ashharrison90 ignore that, it's not required anyway :)

**Comment by ashharrison90** [comment]
ok i'll wait for @ifrost's rereview and suggest we continue the `ToolbarButton` discussion in slack? i'll post something in `#grafana-navigation` now.

**Review by ifrost — APPROVED** [review]
> @ifrost thanks 🙌 i've fixed the bug with `Live` not appearing

Looks good now! Thanks! 🙌

> `ToolbarButton`s not having borders around them when `topnav` is enabled [was a conscious choice](https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/ToolbarButton/ToolbarButton.tsx#L193) and isn't a result of this PR (turning on `topnav` in main should show the same behaviour). 
> **but** pretty sure buttons in the top bar were the only consideration, not sure how we can handle `ToolbarButton`s being used in other places gracefully 🤔

I think it looks okay, but still it's a change so just wanted to double check where is it coming from. Something to think about when we tackle https://github.com/grafana/grafana/issues/54751

**Pull request merged**